### PR TITLE
Modify mgr-influx module database check to not require admin privileges

### PR DESCRIPTION
**Pull request by bmeekhof**
- existing check tried to list all DB and fails even if DB exists if user is not admin level
- still tries to create database if not found and user has privs
- modified measurement for daemon stats to be ceph_daemon_stats (was _osd_)

Signed-off-by: Benjeman Meekhof <bmeekhof@umich.edu>

### CODE REVIEW

File: src/pybind/mgr/influx/module.py [hunk 1]
```diff
@@ -82,7 +83,7 @@ def get_daemon_stats(self):
                 value = counter_info['value']
 
                 data.append({
-                    "measurement": "ceph_osd_stats",
```
Comment (jcsp) [inline src/pybind/mgr/influx/module.py]: This change appears to be unrelated to the commit message -- if it's intentional, please separate it out into a different commit.

File: src/pybind/mgr/influx/module.py [hunk 2]
```diff
@@ -112,13 +113,15 @@ def send_to_influx(self):
...
-        databases_avail = client.get_list_database()
-        if database not in databases_avail:
-            self.log.info("Creating database '{0}'".format(database))
-            client.create_database(database)
 
-        client.write_points(self.get_df_stats(), 'ms')
-        client.write_points(self.get_daemon_stats(), 'ms')
+	# using influx client get_list_database requires admin privs, instead we'll catch the not found exception and inform the user if db can't be created
+	try:
+        	client.write_points(self.get_df_stats(), 'ms')
```
Comment (jcsp) [inline src/pybind/mgr/influx/module.py]: Please use 4-space indentation (no tabs) for python code.

File: src/pybind/mgr/influx/module.py [hunk 3]
```diff
@@ -112,13 +113,15 @@ def send_to_influx(self):
...
+        	client.write_points(self.get_daemon_stats(), 'ms')
+	except InfluxDBClientError as e:
+                if e.code == 404:
+                        self.log.info("Database '{0}' not found, trying to create (requires admin privs).  You can also create manually and grant write privs to user '{1}'".format(database,username))
+                        client.create_database(database)
```
Comment (jcsp) [inline src/pybind/mgr/influx/module.py]: Should also do this:
```
else:
    raise
```

So that if the error is anything other than 404, we're not silently dropping it.

File: src/pybind/mgr/influx/module.py [hunk 4]
```diff
@@ -112,13 +113,15 @@ def send_to_influx(self):
...
 
-        client.write_points(self.get_df_stats(), 'ms')
-        client.write_points(self.get_daemon_stats(), 'ms')
+	# using influx client get_list_database requires admin privs, instead we'll catch the not found exception and inform the user if db can't be created
+	try:
+        	client.write_points(self.get_df_stats(), 'ms')
+        	client.write_points(self.get_daemon_stats(), 'ms')
+	except InfluxDBClientError as e:
+                if e.code == 404:
+                        self.log.info("Database '{0}' not found, trying to create (requires admin privs).  You can also create manually and grant write privs to user '{1}'".format(database,username))
```
Comment (jcsp) [inline src/pybind/mgr/influx/module.py]: Please could you update the docs in doc/mgr/influx.rst to explain that the user must either have admin privs, or that the database must have been precreated

**Comment by bmeekhof** [comment]
@jcsp requested changes made, thanks!
